src/java/fr/paris/lutece/portal/service/download/IFileDownloadProvider.java

```diff
@@ -0,0 +1,39 @@
+package fr.paris.lutece.portal.service.download;
+
+import javax.servlet.http.HttpServletRequest;
+
+import fr.paris.lutece.portal.business.file.File;
+import fr.paris.lutece.portal.service.message.SiteMessageException;
+import fr.paris.lutece.portal.service.util.AppPropertiesService;
+
+/**
+ *  Provider for download of {@link File}
+ */
+public interface IFileDownloadProvider
+{
+    String getProviderName( );
+    
+    /**
+     * Generates an encrypted download url for a {@link File}.
+     * @param fileDownloadData
+     * @return
+     */
+    String getDownloadUrl( FileDownloadData fileDownloadData );
+    
+    /**
+     * Decrypts download parameters and get the file to download
+     * @param request
+     * @return
+     * @throws SiteMessageException
+     */
+    File getFile( HttpServletRequest request ) throws SiteMessageException;
```

Comment:
The fact that this method can return `null` should be documented (or it should be made to never return `null` and throw instead